src/components/NewClusterModal.js

```diff
@@ -42,217 +43,428 @@ const imageValidationRegexp = /^[A-Za-z0-9]+[\w./-]+(?::\w[\w.-]+)?(?:@[\w+.-]+:
+      sparkMode: cloudService === cloudServices.GCE ? false : numberOfWorkers === 0 ? 'master' : 'cluster',
+      ...currentConfig,
+      masterDiskSize: currentCluster?.runtimeConfig?.masterDiskSize || currentCluster?.runtimeConfig?.diskSize || DEFAULT_DISK_SIZE,
+      numberOfWorkers: numberOfWorkers || 2,
+      deleteDiskSelected: false,
+      upgradeDiskSelected: false,
+      simplifiedForm: !currentCluster
     }
   }
 
-  getRuntimeConfig(isNew = false) {
-    return formatRuntimeConfig({
-      cloudService: !!this.state.sparkMode ? 'DATAPROC' : 'GCE',
-      isNew,
-      ..._.pick(
-        ['numberOfWorkers', 'masterMachineType', 'masterDiskSize', 'workerMachineType', 'workerDiskSize', 'numberOfPreemptibleWorkers'],
-        this.state)
-    })
+  makeWorkspaceObj() {
+    const { namespace, name } = this.props
+    return { workspace: { namespace, name } }
   }
 
-  deleteCluster() {
-    const { currentCluster } = this.props
-    const { googleProject, runtimeName } = currentCluster
-
-    return Ajax().Clusters.cluster(googleProject, runtimeName).delete()
+  getCurrentCluster() {
+    const { clusters } = this.props
+    return currentCluster(clusters)
   }
 
-  createCluster() {
-    const { namespace, onSuccess, currentCluster } = this.props
-    const { jupyterUserScriptUri, selectedLeoImage, customEnvImage } = this.state
-    onSuccess(Promise.all([
-      Ajax().Clusters.cluster(namespace, Utils.generateClusterName()).create({
-        runtimeConfig: this.getRuntimeConfig(true),
-        toolDockerImage: selectedLeoImage === CUSTOM_MODE || selectedLeoImage === PROJECT_SPECIFIC_MODE ? customEnvImage : selectedLeoImage,
-        labels: { saturnIsProjectSpecific: `${selectedLeoImage === PROJECT_SPECIFIC_MODE}` },
-        ...(jupyterUserScriptUri ? { jupyterUserScriptUri } : {})
-      }),
-      !!currentCluster && this.deleteCluster()
-    ]))
+  getCurrentPersistentDisk() {
+    const currentCluster = this.getCurrentCluster()
+    const { clusters, persistentDisks } = this.props
+    const id = currentCluster?.runtimeConfig.persistentDiskId
+    const attachedIds = _.without([undefined], _.map(cluster => cluster.runtimeConfig.persistentDiskId, clusters))
+    return id ?
+      _.find({ id }, persistentDisks) :
+      _.last(_.sortBy('auditInfo.createdDate', _.filter(({ id }) => !_.includes(id, attachedIds),
+        _.filter(disk => disk.status !== 'Deleting', persistentDisks))))
   }
 
-  updateCluster(isStopRequired = false) {
-    const { currentCluster, onSuccess } = this.props
-    const { googleProject, runtimeName } = currentCluster
-
-    if (isStopRequired) {
-      notify('info', 'To be updated, your runtime will now stop, and then start. This will take 3-5 minutes.')
-    }
-
-    return onSuccess(
-      Ajax().Clusters.cluster(googleProject, runtimeName).update({
-        runtimeConfig: this.getRuntimeConfig()
+  /**
+   * Transform the new environment config into the shape of runtime config
```

Comment:
Ah, thank you for giving me some guidance! I don't think I've ever settled on a personal preference for the phrasing of doc comments like this. I like the notion of making sure it sounds like a description rather than a command/request/TODO.